src/AutoFixture.AutoEF.Tests/NavigationPropertyRequestSpecificationTest.cs

```diff
@@ -1,72 +1,105 @@
+        {
+            sut.Invoking(s => s.IsSatisfiedBy(null))
+                .ShouldThrow<ArgumentNullException>();
+        }
+
+        [Theory, AutoNSub]
+        public void ReturnsFalseWhenDeclaringTypeNotRegistered([Frozen] IEntityTypesProvider typesProvider, SUT sut)
+        {
+            typesProvider.GetTypes().Returns(new[] { typeof(Bar) });
+            var property = typeof (Foo).GetProperty("Bar");
+
+            var result = sut.IsSatisfiedBy(property);
+
+            result.Should().BeFalse();
+        }
+
+        [Theory, AutoNSub]
+        public void ReturnsFalseWhenPropertyTypeNotRegistered([Frozen] IEntityTypesProvider typesProvider, SUT sut)
+        {
+            typesProvider.GetTypes().Returns(new[] { typeof(Foo) });
+            var property = typeof(Foo).GetProperty("Bar");
+
+            var result = sut.IsSatisfiedBy(property);
+
+            result.Should().BeFalse();
+        }
+
+        [Theory, AutoNSub]
+        public void ReturnsTrueWhenBothTypesRegistered([Frozen] IEntityTypesProvider typesProvider, SUT sut)
+        {
+            typesProvider.GetTypes().Returns(new[] { typeof(Foo), typeof(Bar) });
+            var property = typeof(Foo).GetProperty("Bar");
+
+            var result = sut.IsSatisfiedBy(property);
+
+            result.Should().BeTrue();
+        }
+
+        [Theory, AutoNSub]
+        public void ReturnsTrueWhenPropertyIsCollection([Frozen] IEntityTypesProvider typesProvider, SUT sut)
+        {
+            typesProvider.GetTypes().Returns(new[] { typeof(Qux), typeof(Bar) });
+            var property = typeof(Bar).GetProperty("Quxes");
+
+            var result = sut.IsSatisfiedBy(property);
+
+            result.Should().BeTrue();
+        }
+
+        [Theory, AutoNSub]
+        public void ReturnsFalseWhenDeclaringTypeNotRegisteredWhenTableNameId([Frozen] IEntityTypesProvider typesProvider, SUT sut)
+        {
+            typesProvider.GetTypes().Returns(new[] { typeof(Boo) });
+            var property = typeof(Far).GetProperty("Boo");
+
+            var result = sut.IsSatisfiedBy(property);
+
+            result.Should().BeFalse();
+        }
+
+        [Theory, AutoNSub]
+        public void ReturnsFalseWhenPropertyTypeNotRegisteredWhenTableNameId([Frozen] IEntityTypesProvider typesProvider, SUT sut)
+        {
+            typesProvider.GetTypes().Returns(new[] { typeof(Far) });
+            var property = typeof(Far).GetProperty("Boo");
+
+            var result = sut.IsSatisfiedBy(property);
+
+            result.Should().BeFalse();
+        }
+
+        [Theory, AutoNSub]
+        public void ReturnsTrueWhenBothTypesRegisteredWhenTableNameId([Frozen] IEntityTypesProvider typesProvider, SUT sut)
+        {
+            typesProvider.GetTypes().Returns(new[] { typeof(Far), typeof(Boo) });
+            var property = typeof(Far).GetProperty("Boo");
+
+            var result = sut.IsSatisfiedBy(property);
+
+            result.Should().BeTrue();
+        }
+    }

```

Comment:
I believe this file should remain unchanged from the original, as the tests only concern `NavigationPropertyRequestSpecification` which was not modified